tests/docker_run_tests.rs

```diff
@@ -245,6 +247,63 @@ fn run_postgres() -> Container {
     }
 }
 
+fn run_mysql() -> Container {
+    let mut docker_cmd = Command::new("docker");
+
+    let hash = Uuid::new_v4().to_string();
+    let container_name = format!("mysql-{}", hash);
+    let password = hash;
+    // run
+    docker_cmd.arg("run");
+
+    // Set Needed Envvars
+    docker_cmd
+        .arg("-e")
+        .arg(format!("MYSQL_ROOT_PASSWORD={}", &password))
+        .arg("-e")
+        .arg(format!("MYSQL_PASSWORD={}", &password))
+        .arg("-e")
+        .arg("MYSQL_USER=mysql")
+        .arg("-e")
+        .arg("MYSQL_DATABASE=mysql");
+
+    // Run detached
+    docker_cmd.arg("-d");
+
+    // attach name
+    docker_cmd.arg("--name").arg(container_name.clone());
+
+    // Assign image
+    docker_cmd.arg(MYSQL_IMAGE);
+
+    // Run the command
+    docker_cmd
+        .spawn()
+        .unwrap()
+        .wait()
+        .context("starting mysql")
+        .unwrap();
+
+    // MySQL starts listening for connections after it has initialised its default database
+    // give it enough time to start before we continue with starting the application
+    // TODO: swap this for a healthceck + retry-loop because the arbitrary delay can lead to flakey tests
+    thread::sleep(Duration::new(10, 0));
```

Comment:
@coffee-cup thanks for the feedback - implemented the tests. But MySQL is being a bit annoying here, will address the TODO and ping you once it's ready for review.

Thinking of running a retry-loop with timeout (say 30s) of a command to check if MySQL is ready or not inside this function.

Reply:
actually - looking at the runtimes, it seems like some of the docker tests take a few mins to complete. Could also add some margin to the sleep  (eg: make it 20s) and leave it as-is to reduce complexity if we're ok with that.

Reply:
whoops missed the unit test